### PR TITLE
Add comments to template

### PR DESCRIPTION
**Pull request by agnieszka-m**

### CODE REVIEW

File: pipelines/DenseDocSearch.yaml [hunk 1]
```diff
@@ -1,34 +1,35 @@
...
-      embedding_model: sentence-transformers/multi-qa-mpnet-base-dot-v1
+      embedding_model: sentence-transformers/multi-qa-mpnet-base-dot-v1 #model optimized for semantic search
       model_format: sentence_transformers
-      top_k: 20
-  - name: FileTypeClassifier
+      top_k: 20 #the number of results to return
+  - name: FileTypeClassifier #routes files based on their extension to appropriate converters
     type: FileTypeClassifier
-  - name: TextConverter
+  - name: TextConverter #converts files into documents
```
Comment (tholor) [inline pipelines/DenseDocSearch.yaml:18]: "converts .txt files into documents"

File: pipelines/DenseDocSearch.yaml [hunk 2]
```diff
@@ -1,34 +1,35 @@
...
-# All nodes that we need for our pipelines
+# This section defines nodes that we want to use in our pipelines
 components:
   - name: DocumentStore
-    type: DeepsetCloudDocumentStore
-  - name: Retriever
-    type: EmbeddingRetriever
+    type: DeepsetCloudDocumentStore #the only supported document store in deepset Cloud
+  - name: Retriever #selects the most relevant documents from the document store
+    type: EmbeddingRetriever #uses a Transformer model to encode the document and the query
```
Comment (tholor) [inline pipelines/DenseDocSearch.yaml:10]: "uses one Transformer model ..." 
(to clarify the contrast to later DPR models which use two transformer models ...)

File: pipelines/QuestionAnswering_de.yaml [hunk 3]
```diff
@@ -2,42 +2,42 @@
...
 components:
   - name: DocumentStore
-    type: DeepsetCloudDocumentStore
+    type: DeepsetCloudDocumentStore #the only supported document store in deepset Cloud
     params:
-      similarity: cosine
-  # A fast retriever that finds the top 20 candidate documents
-  - name: Retriever
-    type: EmbeddingRetriever
+      similarity: cosine #recommended for BERT models
```
Comment (tholor) [inline pipelines/QuestionAnswering_de.yaml:10]: "#recommended for sentence transformer models"

**Review by tholor — APPROVED** [review]
Nice! Added only minor comments. Feel free to merge once those are addressed

**Comment by erendabanlioglu** [comment]
@agnieszka-m I updated the comments. For 2nd suggestion I know "a Transformer model" already means "one" model but to emphasize I think we can agree on that.

**Comment by agnieszka-m** [comment]
thanks guys!

**Pull request merged**